src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/NuGetTargetFrameworkMonikerComparer.cs

```diff
@@ -112,14 +112,17 @@ public bool TryMerge(TargetFrameworkMoniker tfm1, TargetFrameworkMoniker tfm2, [
                 _ => null,
             };
 
-            var newFramework = (platform, version) switch
+            var frameworkName = (maxFramework.Framework, maxFramework.Version) switch
             {
-                (string p, Version v) => new NuGetFramework(maxFramework.Framework, maxFramework.Version, p, v),
-                (string p, null) => new NuGetFramework(maxFramework.Framework, maxFramework.Version, p, new Version(0, 0, 0, 0)),
-                _ => new NuGetFramework(maxFramework.Framework, maxFramework.Version),
+                (".NETCoreApp", Version v) when v >= TargetFrameworkMoniker.Net50.FrameworkVersion => "net",
+                (string f, _) => f
             };
 
-            result = new TargetFrameworkMoniker(newFramework.GetShortFolderName());
+            result = new TargetFrameworkMoniker(frameworkName, maxFramework.Version)
```

Comment:
Not related to this change, but I think we missed a scenario in the `GetPlatform` method.

If you merge *net5.0-windows10.0.19041.0* and *net6.0-windows10.0.18362.0* 

I think you should get *net6.0-windows**10.0.19041.0***

Reply:
I believe it does. Here's the relevant test:

https://github.com/dotnet/upgrade-assistant/blob/cb835e9c4059b0c4f1ffa350e84b905b99ea8486/tests/components/Microsoft.DotNet.UpgradeAssistant.MSBuild.Tests/TargetFrameworkIdentifierTests.cs#L91